modules/rostests/apitests/wscript/wsf.c

```diff
@@ -0,0 +1,166 @@
+ * COPYRIGHT:   ReactOS Team
+ */
+
+#include <apitest.h>
+#include <windows.h>
+#include <shlwapi.h>
+
+#define MYGUID "{898AC78E-BFC7-41FF-937D-EDD01E666707}"
+
+static DWORD getregdw(HKEY hKey, LPCSTR sub, LPCSTR name, DWORD *out, DWORD defval)
+{
+    DWORD size = sizeof(*out);
+    *out = 0;
+    LRESULT e = SHGetValueA(hKey, sub, name, NULL, out, &size);
+    if (e)
+        *out = defval;
+    return e;
+}
+
+static BOOL makestringfile(LPWSTR path, LPCSTR ext, LPCSTR string, const BYTE *map)
+{
+    UINT cch = GetTempPathW(MAX_PATH, path);
+    UINT16 i = 0;
+    while (++i)
+    {
+        HANDLE hFile;
+        wsprintfW(path + cch, L"~%u.%hs", i, ext ? ext : "tmp");
+        hFile = CreateFileW(path, GENERIC_WRITE, FILE_SHARE_READ, 0, CREATE_NEW, 0, NULL);
+        if (hFile != INVALID_HANDLE_VALUE)
+        {
+            BOOL succ = TRUE;
+            for (; *string && succ; ++string)
+            {
+                BYTE ch = *string;
+                DWORD j;
+                for (j = 0; map && map[j + 0]; j += 2)
+                {
+                    if (ch == map[j + 0])
+                        ch = map[j + 1];
+                }
+                succ = WriteFile(hFile, &ch, 1, &j, NULL);
+            }
+            CloseHandle(hFile);
+            return succ;
+        }
+    }
+    return FALSE;
+}
+
+static DWORD runscriptfile(LPCWSTR path, LPCWSTR engine)
+{
+    STARTUPINFOW si;
+    PROCESS_INFORMATION pi;
+    LPCWSTR exe = engine ? engine : L"wscript.exe";
+    WCHAR cmd[MAX_PATH * 2];
+    wsprintfW(cmd, L"\"%s\" //nologo \"%s\"", exe, path);
+    ZeroMemory(&si, sizeof(si));
+    si.cb = sizeof(si);
+    if (CreateProcessW(NULL, cmd, NULL, NULL, FALSE, 0, NULL, NULL, &si, &pi))
+    {
+        DWORD code = ERROR_INTERNAL_ERROR;
+        WaitForSingleObject(pi.hProcess, INFINITE);
+        GetExitCodeProcess(pi.hProcess, &code);
+        CloseHandle(pi.hProcess);
+        CloseHandle(pi.hThread);
+        return code;
+    }
+    return GetLastError();
+}
+
+static DWORD runscript(LPCSTR ext, LPCSTR script, const BYTE *map)
+{
+    WCHAR file[MAX_PATH];
+    DWORD code;
+    if (!makestringfile(file, ext, script, map))
+    {
+        skip("Unable to create script\n");
+        return ERROR_FILE_NOT_FOUND;
+    }
+    code = runscriptfile(file, NULL);
+    DeleteFileW(file);
+    return code;
+}
+
+static void test_defaultscriptisjs()
+{
+    LPCSTR script = ""
+    "<job>"
+    "<script>" /* No language attribute should default to Javascript */
+    "var x = 42;"
+    "WScript.Quit(x);"
+    "</script>"
+    "</job>";
+
+    ok(runscript("wsf", script, NULL) == 42, "Script failed\n");
+}
+
+static void test_simplevb()
+{
+    LPCSTR script = ""
+    "<job>"
+    "<script language=\"VBScript\">"
+    "Dim x\n"
+    "x = 42\n"
+    "WScript.Quit x\n"
+    "</script>"
+    "</job>";
+
+    ok(runscript("wsf", script, NULL) == 42, "Script failed\n");
+}
+
+static void test_defpackagejob()
+{
+    LPCSTR script = ""
```

Comment:
```suggestion
    static LPCSTR script = ""
```

Reply:
No, that makes no sense.

You perhaps mean `static const char script[] = "...` but string literals are already static globals.

Reply:
There is still a difference:
https://godbolt.org/z/jefaW6fvv
Check how assembly changes in either Func1 or Func2 if you add or remove the "static".
(and I could have suggested `static const LPCSTR script` to make the value of the pointer const and also put it in the static globals, not just the string bytes themselves.)

Reply:
I cannot believe we are going to have this discussion!

```
int foo = 1337;
SetLastError(foo);
static int bar = 42;
SetLastError(bar);
```
Are these two the same? No, of course not.

https://godbolt.org/z/TaGr4cbY8

With optimizations on, all variants turn into the same thing, the only difference is that `static LPCSTR` is basically a global variable that you are allocating for no reason.